monkey/monkey_island/cc/ui/src/components/report-components/ransomware/AttackSection.tsx

```diff
@@ -0,0 +1,102 @@
+import React, {ReactElement, ReactFragment, useEffect, useState} from 'react';
+import IslandHttpClient from '../../IslandHttpClient';
+import {FileEncryptionTable, TableRow} from './FileEncryptionTable';
+import NumberedReportSection from './NumberedReportSection';
+import LoadingIcon from '../../ui-components/LoadingIcon';
+
+const ATTACK_DESCRIPTION = 'After the attacker or malware has propagated through your network, \
+                            your data is at risk on any machine the attacker can access. It can be \
+                            encrypted and held for ransomware, exfiltrated, or manipulated in \
+                            whatever way the attacker chooses.'
+const HOSTNAME_REGEX = /^(.* - )?(\S+) :.*$/;
+
+function AttackSection(): ReactElement {
+  const [tableData, setTableData] = useState(null);
+
+  useEffect(() => {
+    IslandHttpClient.get('/api/telemetry?telem_category=file_encryption')
+      .then(resp => setTableData(processTelemetry(resp.body)));
+  }, []);
+
+
+  if (tableData == null) {
+      return <LoadingIcon />
+  }
+
+  return (
+    <NumberedReportSection
+      index={3}
+      title='Attack'
+      description={ATTACK_DESCRIPTION}
+      body={getBody(tableData)}
+    />
+  );
+}
+
+function getBody(tableData): ReactFragment {
+  return (
+    <>
+      <p>Infection Monkey has encrypted <strong>{tableData.length} files</strong> on your network:</p>
+      <FileEncryptionTable tableData={tableData} />
+    </>
+  );
+}
+
+function processTelemetry(telemetry): Array<TableRow> {
+  // Sort ascending so that newer telemetry records overwrite older ones.
+  sortTelemetry(telemetry);
```

Comment:
Maybe sort according to the hostname, as the user won't be able to sort according to the time again. Sorting in such a large table becomes a prominent feature.

Reply:
This sort is necessary to avoid https://github.com/guardicore/monkey/pull/1363#discussion_r676618693